staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -221,6 +221,9 @@ func (p *Parser) parseRecursive(cur *ListNode) error {
 	cur.append(newRecursive())
 	if r := p.peek(); isAlphaNumeric(r) {
 		return p.parseField(cur)
+	} else if r == '.' {
+		// one or more periods following the recursive decent operator .. is invalid
+		return fmt.Errorf("invalid recursive decent")
```

Comment:
it looks like returning an error here will make the CR handler fall back to default printing:

https://github.com/kubernetes/kubernetes/blob/607c5daabd4204fcebe90f983ba06edfcc6fc4e5/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L751-L754

Do we need to do something to detect this and make it more visible in the CRD status?

Reply:
cc @sttts

Reply:
We could, but IMO that's a separate issue that already exists today for any invalid jsonpath expression in a CRD column.

Also, it's probably worth mentioning that we could swallow all the extra dots immediately following 2 dots (instead of returning an error) if we think that's better.

Reply:
In addition, we could also validate the jsonpath in apiserver.

Reply:
tightening validation introduces skew problems with existing persisted data. let's not add that without further discussion.